packages/react-liveness/src/components/FaceLivenessDetector/service/utils/sessionInformation.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  FaceMovementServerChallenge,
+  FaceMovementAndLightServerChallenge,
+  ServerChallenge,
+  SessionInformation as ServerSessionInformation,
+} from '@aws-sdk/client-rekognitionstreaming';
+import {
+  FaceMovementAndLightChallenge,
+  FaceMovementChallenge,
+  SessionInformation,
+} from '../types';
+import {
+  FACE_MOVEMENT_CHALLENGE,
+  FACE_MOVEMENT_AND_LIGHT_CHALLENGE,
+} from './constants';
+
+export const isFaceMovementAndLightChallenge = (
+  value: unknown
+): value is FaceMovementAndLightChallenge => {
+  return (
+    (value as SessionInformation)?.Challenge?.name ===
+    'FaceMovementAndLightChallenge'
+  );
+};
+
+export const isFaceMovementChallenge = (
+  value: unknown
+): value is FaceMovementChallenge => {
+  return (
+    (value as SessionInformation)?.Challenge?.name === 'FaceMovementChallenge'
+  );
+};
+
+export const isFaceMovementAndLightServerChallenge = (
+  value: unknown
+): value is FaceMovementAndLightServerChallenge => {
+  return !!(value as ServerChallenge.FaceMovementAndLightChallengeMember)
+    ?.FaceMovementAndLightChallenge;
+};
+
+export const isFaceMovementServerChallenge = (
+  value: unknown
+): value is FaceMovementServerChallenge => {
+  return !!(value as ServerChallenge.FaceMovementChallengeMember)
+    ?.FaceMovementChallenge;
+};
+
+export const createSessionInfoFromServerSessionInformation = (
+  serverSessionInformation: ServerSessionInformation
+): SessionInformation => {
+  let challenge: FaceMovementAndLightChallenge | FaceMovementChallenge;
+  if (
+    isFaceMovementAndLightServerChallenge(serverSessionInformation.Challenge)
+  ) {
+    challenge = {
+      ...serverSessionInformation.Challenge.FaceMovementAndLightChallenge,
+      name: FACE_MOVEMENT_AND_LIGHT_CHALLENGE.Type,
```

Comment:
What benefit is there to doing this as opposed to writing like a utility function which gets the challenge typename?

Reply:
In a decent amount of cases (like stuff involving challengeConfig and ovalDetails) the challenge type doesn't matter bc the attributes exist on both. Removing the `FaceMovementAndLightChallenge`/`FaceMovementChallenge` property saves us from having to check the type any time we access an attribute. like we can do 
`Challenge.ChallengeConfig` instead of 
`isFaceMovementAndLightChallenge ? Challenge.FaceMovementAndLightChallenge.ChallengeConfig ? Challenge.FaceMovementChallenge.ChallengeConfig`

Also was thinking it would make the code more extensible and easier to refactor if/when new challenges are added